### PR TITLE
feat: TLS for Jetstream 

### PR DESCRIPTION
**Pull request by juliev0**
Fixes: #1797 

Includes both client->server and intra-cluster

Uses self-signed certificates

Checklist:

* [ ] My organization is added to [USERS.md](https://github.com/argoproj/argo-events/blob/master/USERS.md).

### CODE REVIEW

File: controllers/eventbus/installer/assets/jetstream/nats.conf [hunk 1]
```diff
@@ -29,6 +29,12 @@ cluster {
   routes: [{{.Routes}}]
   cluster_advertise: $CLUSTER_ADVERTISE
   connect_retries: 120
+
+  tls {
+    cert_file: "/etc/nats-config/cluster-server-cert.pem"
+    key_file:  "/etc/nats-config/cluster-server-key.pem"
+    ca_file:   "/etc/nats-config/cluster-ca-cert.pem"
+  }
```
Comment (juliev0) [inline controllers/eventbus/installer/assets/jetstream/nats.conf:37]: In some ways this belongs in server-auth.conf. I tried adding a second "cluster" section into there but I don't think NATS can handle the two sections.

File: common/leaderelection/leaderelection.go [hunk 2]
```diff
@@ -66,10 +69,11 @@ func NewEventBusElector(ctx context.Context, eventBusConfig eventbusv1alpha1.Bus
 			logger.Fatal("Eventbus auth config file changed, exiting..")
 		})
 		auth = &eventbuscommon.Auth{
-			Strategy:    *eventBusAuth,
-			Crendential: cred,
+			Strategy:   *eventBusAuth,
+			Credential: cred,
```
Comment (whynowy) [inline common/leaderelection/leaderelection.go:73]: 👍

File: controllers/eventbus/installer/jetstream.go [hunk 3]
```diff
@@ -498,41 +489,117 @@ func (r *jetStreamInstaller) createSecrets(ctx context.Context) error {
...
 		}
-		r.logger.Infow("deleted malformed nats server auth secret successfully")
-	}
 
-	if oldServerObjExisting {
-		if err := r.client.Delete(ctx, oldCObj); err != nil {
-			return fmt.Errorf("failed to delete malformed nats client auth secret, err: %w", err)
+		// Generate TLS self signed certificate for Jetstream cluster nodes: includes TLS private key, certificate, and CA certificate
+		clusterNodeHosts := []string{}
+		for i := 0; i < r.eventBus.Spec.JetStream.GetReplicas(); i++ {
```
Comment (whynowy) [inline controllers/eventbus/installer/jetstream.go:532]: Is it able to do "*.xxx"? Updating replicas will not regenerate the CERT.

File: controllers/eventbus/installer/assets/jetstream/server-auth.conf [hunk 4]
```diff
@@ -10,3 +10,9 @@ accounts: {
     ]
   }
 }
+
+tls {
```
[the remark this answers is not in the document]
Comment (juliev0) [reply controllers/eventbus/installer/assets/jetstream/server-auth.conf:17]: Are you looking above? The other one is inside of "cluster{}" (it's doing between cluster nodes)

File: controllers/eventbus/installer/jetstream.go (hunk 3 above)
Comment (juliev0) [reply controllers/eventbus/installer/jetstream.go:532]: Good thinking. Let me see.

File: controllers/eventbus/installer/assets/jetstream/server-auth.conf (hunk 4 above)
[the remark this answers is not in the document]
Comment (whynowy) [reply controllers/eventbus/installer/assets/jetstream/server-auth.conf:17]: I see.

File: controllers/eventbus/installer/jetstream.go (hunk 3 above)
Comment (juliev0) [reply controllers/eventbus/installer/jetstream.go]: Great, using a wildcard in the first position seems to work: 
`"fmt.Sprintf("*.%s.%s.svc.cluster.local", generateJetStreamServiceName(r.eventBus), r.eventBus.Namespace)"`

**Review by whynowy — APPROVED** [review]
LGTM! Good work!

**Pull request merged**